### PR TITLE
flutter min version test

### LINKED ISSUE #636: e2e tests should run on min. supported sdk versions
running on the min. supported SDK version, we'd be able to catch back-compatibility during CI run and avoid #634 similar issues

### LINKED ISSUE #962: Set min. Flutter version to `2.0.0`
https://github.com/getsentry/sentry-dart/blob/53b6921e2b215fb46eb9dab8f9720d012c3ee15a/flutter/pubspec.yaml#L10

https://developers.googleblog.com/2021/03/announcing-flutter-2.html
https://medium.com/dartlang/announcing-dart-2-12-499a6e689c87
https://twitter.com/timsneath/status/1551455025435590656

The current GA version is 3.x already.
Most people use 3.x.
Supporting older APIs have a cost, such as [bundle size](https://twitter.com/mraleph/status/1551440298135506945) increase and casting code to `dynamic`.
People can still use the older version of the Sentry Flutter SDK.

If you are using Flutter 1.x, please let us know.

### PR DESCRIPTION
**Pull request by marandaneto**
## :scroll: Description
TODO: changelog
_#skip-changelog_

Bump Flutter to min 2.0.0
Add CI step that compiles using Flutter's min. supported version now 2.0.0 and Dart 2.12.0
Fix `hashCode` issue.
Add dependency for `logging` and `dio` so they are compiled against the min. version as well.

## :bulb: Motivation and Context
Fixes https://github.com/getsentry/sentry-dart/issues/962
Fixes https://github.com/getsentry/sentry-dart/issues/636

## :pencil: Checklist

- [X] I reviewed submitted code
- [X] I added tests to verify changes
- [ ] I updated the docs if needed
- [X] All tests passing
- [ ] No breaking changes

### CODE REVIEW

**Comment by marandaneto** [comment]
@ueman found one more issue:

>     ../dart/lib/src/client_reports/client_report_recorder.dart:46:30: Error: Method not found: 'Object.hash'.
>       int get hashCode => Object.hash(reason, category);

File: .github/workflows/min_version_test.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+name: min version test
+on:
+  push:
+    branches:
+      - main
+      - release/**
+  pull_request:
+    paths-ignore:
+      - 'logging/**'
+      - 'dio/**'
```
Comment (marandaneto) [inline .github/workflows/min_version_test.yml]: Maye this should be gone.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,11 @@
 # Changelog
 
+## Unreleased
+
+### Features
+
+* Bump Flutter's min. supported version from 1.17.0 to 2.0.0 ([#966](https://github.com/getsentry/sentry-dart/pull/966))
```
Comment (ueman) [inline CHANGELOG.md:7]: Maybe add, that this shouldn't break anyone anyway?

**Review by brustolin — APPROVED** [review]
LGTM

**Pull request merged**